### PR TITLE
Hard link archive

### PR DESCRIPTION
**Pull request by benjwadams**
Archives files using hard links rather than copying, saving a non-trivial amount of space.

### CODE REVIEW

**Comment by benjwadams** [comment]
@greenteadigital, please review.  I can't add you as a reviewer.

File: scripts/archive_datasets.py [hunk 1]
```diff
@@ -67,9 +67,9 @@ def make_copy(filepath):
     if not os.path.exists(target):
         logger.info("Creating archive dataset")
         try:
-            shutil.copyfile(source, target)
-        except IOError:
-            logger.exception("Could not copy file {}".format(source))
+            os.link(source, target)
+        except (IOError, OSError):
+            logger.exception("Could not hard link to file {}".format(source))
```
Comment (greenteadigital) [inline scripts/archive_datasets.py:72]: How about
`logger.exception(f"Could not hard link to file { source }")`
just a personal preference for f-strings over string.format()

**Comment by greenteadigital** [comment]
I'm not able to merge this Ben. Feel free to do so.

**Comment by mwengren** [comment]
@benjwadams I sent a request to add @greenteadigital to the IOOS organization and team that has access to this repository.

**Comment by mwengren** [comment]
Also @benjwadams you should have the ability to add @greenteadigital to any IOOS teams you have maintainer role for: https://github.com/orgs/ioos/teams.

**Pull request merged**